### ISSUE TITLE
Help regarding sections in new configuration

### ISSUE THREAD

**Issue by rheaditi** [styleguidist/react-styleguidist#1139]
### tl;dr

Hey guys! Just upgraded to `7.3.8` from `7.0.20` and something went wrong with my sections! 😅 

### The issue

We were using a `sections` configuration similar to this:

```js
{
  sections: [{
     name: 'Typography',
     sections: [
       utils.getComponentSection('Heading'),
       utils.getComponentSection('Text'),
     ],
   },
  // ...
  ],
 pagePerSection: true,
}
```

Where `getComponentSection` would return

```js
{
  name: 'Heading',
  content: 'src/components/Heading/Heading.md',
  component: 'src/components/Heading/Heading.ts',
}
```

With the above configuration, `Typography` would render as a page with all its subsections in the same page, and if you would click on `Heading` or `Text` under `Typography`, they would render only that section's content due to `pagePerSection: true`. This worked really well for us. 🙂 

However after the upgrade the above configuration does not seem to be working. We tried playing around with the `sectionDepth` field but when we applied any non-zero value to `sectionDepth`, the `Typography` section would render a blank page.

Can you help us achieve the outcome of the previous configuration with the new changes in version `7.3.8`? We're finding it a bit challenging to understand how `sectionDepth` works. 

**Or**, is there a way where we can have a section heading which is **_not_** a link; rather it's just there to categorize components in a meaningful manner?

Also, thanks for a great tool! It has made maintaining our styleguide so much easier 🙂👍
